move/its/sources/service.move

```diff
@@ -273,20 +287,45 @@ module its::service {
     /// Decode an approved call and check that the source chain is trusted.
     fun decode_approved_message(self: &mut ITS, approved_message: ApprovedMessage): (String, vector<u8>) {
         let (
-            source_chain,
+            mut source_chain,
             _,
             source_address,
-            payload
+            mut payload
         ) = self.channel_mut().consume_approved_message(approved_message);
 
         assert!(self.is_trusted_address(source_chain, source_address), EUntrustedAddress);
 
+        let mut reader = abi::new_reader(payload);
+        if (reader.read_u256() == MESSAGE_TYPE_RECEIVE_FROM_HUB) {
+            assert!(source_chain.into_bytes() == ITS_HUB_CHAIN_NAME, ESenderNotHub);
+            source_chain = ascii::string(reader.read_bytes());
```

Comment:
Missing check for whether original source chain is registered correctly
https://github.com/axelarnetwork/interchain-token-service/blob/main/contracts/InterchainTokenService.sol#L978-L981

Reply:
fixed